collect_app/src/main/java/org/odk/collect/android/dao/InstancesDao.java

```diff
@@ -145,6 +146,41 @@ public void deleteInstancesDatabase() {
         Collect.getInstance().getContentResolver().delete(InstanceProviderAPI.InstanceColumns.CONTENT_URI, null, null);
     }
 
+    public void deleteInstancesFromIDs(String ids[]){
+        int count = ids.length;
+        int i = 0 ;
+        while (count > 0) {
+            String[] selectionArgs = null;
+            if (count > ApplicationConstants.SQLITE_MAX_VARIABLE_NUMBER ) {
+                selectionArgs = new String[
+                        ApplicationConstants.SQLITE_MAX_VARIABLE_NUMBER];
+            } else {
+                selectionArgs = new String[count];
+            }
+
+            StringBuilder selection = new StringBuilder();
+            selection.append(InstanceProviderAPI.InstanceColumns.INSTANCE_FILE_PATH + " IN (");
+            int j = 0 ;
+            while (j < selectionArgs.length) {
+                selectionArgs[j] = ids[i];
```

Comment:
This is the assignment that I mention in the other comment. Since you're reassigning it, just pass the list as the parameter instead of casting it.